packages/hash/api/src/model/knowledge/user.model.ts

```diff
@@ -129,35 +223,141 @@ export default class extends EntityModel {
+  async getKratosIdentity(): Promise<KratosUserIdentity> {
+    const kratosIdentityId = this.getKratosIdentityId();
+    const { data: kratosIdentity } = await adminKratosSdk.adminGetIdentity(
+      kratosIdentityId,
+    );
+
+    return kratosIdentity;
+  }
+
+  async getQualifiedEmails(): Promise<QualifiedEmail[]> {
+    const emails: string[] = (this.properties as any)[emailBaseUri];
+
+    const kratosIdentity = await this.getKratosIdentity();
+
+    return emails.map((address) => {
+      const kratosEmail = kratosIdentity.verifiable_addresses?.find(
+        ({ value }) => value === address,
+      );
+
+      if (!kratosEmail) {
+        throw new Error(
+          "Could not find verifiable email address in kratos identity",
+        );
+      }
+
+      const { verified } = kratosEmail;
+
+      return {
+        address,
+        verified,
+        /** @todo: stop hardcoding this */
+        primary: true,
+      };
+    });
+  }
+
   getEmails(): string[] {
     return (this.properties as any)[emailBaseUri];
   }
 
-  getShortname(): string {
+  getShortname(): string | undefined {
     return (this.properties as any)[shortnameBaseUri];
   }
+
+  async updateShortname(
+    _graphApi: GraphApi,
+    _params: { updatedByAccountId: string; updatedShortname: string },
+  ) {
+    /** @todo: re-implement this method */
+    throw new Error("user.updateShortname is not yet re-implemented");
+  }
+
+  getPreferredName(): string | undefined {
+    return (this.properties as any)[preferredNameBaseUri];
+  }
+
+  async updatePreferredName(
+    _graphApi: GraphApi,
+    _params: { updatedByAccountId: string; updatedPreferredName: string },
+  ) {
+    /** @todo: re-implement this method */
+    throw new Error("user.updatePreferredName is not yet re-implemented");
+  }
+
+  getKratosIdentityId(): string {
+    return (this.properties as any)[kratosIdentityIdBaseUri];
+  }
+
+  getAccountId(): string {
+    return (this.properties as any)[accountIdBaseUri];
+  }
+
+  getInfoProvidedAtSignup(): any {
+    throw new Error("user.getInfoProvidedAtSignup is not yet re-implemented");
+  }
+
+  async updateInfoProvidedAtSignup(
+    _graphApi: GraphApi,
+    _params: { updatedByAccountId: string; updatedInfo: any },
+  ) {
+    /** @todo: re-implement this method */
+    throw new Error(
+      "user.updateInfoProvidedAtSignup is not yet re-implemented",
+    );
+  }
+
+  async joinOrg(
+    _graphApi: GraphApi,
+    _params: { org: any; responsibility: string; updatedByAccountId: string },
+  ) {
+    /** @todo: re-implement this method */
+    throw new Error("user.joinOrg is not yet re-implemented");
+  }
+
+  async isMemberOfOrg(
+    _graphApi: GraphApi,
+    _params: { orgEntityId: string },
+  ): Promise<boolean> {
+    /** @todo: re-implement this method */
+    throw new Error("user.isMemberOfOrg is not yet re-implemented");
+  }
+
+  isAccountSignupComplete(): boolean {
+    return !!this.getShortname() && !!this.getPreferredName();
```

Comment:
shouldn't we also check that the accountId exists here?

Reply:
`accountId` is a required property type on user, whereas `shortname` and `preferredName` are optional. This is because `accountId` can be set at the point of creation of the entity, whereas `shortname` and `preferredName` will be provided after the kratos registration flow.

Therefore I don't see the value in checking whether the `accountId` exists here.